docs/learn/test-runner/writing-plugins.md

```diff
@@ -6,9 +6,301 @@ eleventyNavigation:
   order: 70
 ---
 
-...
+Not every use case will be covered by existing plugins 😱
+Therefore if you encounter a situation that requires some custom adjustments you can create a plugin yourself.
+
+## Use case dependencies
+
+We would like to create a tooltip that is responsive.
+Doing something like that from scratch might not be a good idea as there are many edge cases in regards to positioning.
+
+We decided to use a popular package called [popper.js](https://popper.js.org/).
+
+To get started we install it from npm
+
+```bash
+npm i @popperjs/core
+```
+
+then we created a test file in accordance with popper documentation
+
+👉 `test/tooltip.test.js`
+
+```js
+import { expect, fixture, html } from '@open-wc/testing';
+import { createPopper } from '@popperjs/core';
+
+it('can use popper', async () => {
+  const wrapper = await fixture(html`
+    <div>
+      <button id="button" aria-describedby="tooltip">I'm a button</button>
+      <div id="tooltip" role="tooltip">I'm a tooltip</div>
+    </div>
+  `);
+
+  const button = wrapper.querySelector('#button');
+  const tooltip = wrapper.querySelector('#tooltip');
+
+  createPopper(button, tooltip, {
+    placement: 'right',
+  });
+});
+```
+
+when we execute it we hover get an error 🤔
+
+```bash
+$ npm run test
+
+> web-test-runner "test/**/*.test.js" --node-resolve
+
+test/tooltip.test.js:
+
+ ❌ can use popper
+      at: node_modules/@popperjs/core/lib/index.js:81:9
+      ReferenceError: process is not defined
+        at Object.setOptions (node_modules/@popperjs/core/lib/index.js:81:9)
+        at createPopper (node_modules/@popperjs/core/lib/index.js:215:14)
+        at n.<anonymous> (test/tooltip.test.js:25:3)
+```
+
+If you are often using es modules directly in the browser then `ReferenceError: process is not defined` might sound familiar.
+Many libraries sadly decide to publish their code only run-able in node or via a bundler.
+There is simply no `process` variable set within a browser.
```

Comment:
I don't think it's *that* common, I've 

Maybe something like:

Some packages use the global `process.env` variable to check for environment variable. This variable is available in node, but not in the browser.

Reply:
changed